extension/js/common/core/crypto/pgp/openpgp-key.ts

```diff
@@ -196,19 +196,24 @@ export class OpenPGPKey {
     const algoInfo = opgpKey.primaryKey.getAlgorithmInfo();
     const key = keyToUpdate || {} as Key; // if no key to update, use empty object, will get props assigned below
     const encryptionKey = await Catch.undefinedOnException(opgpKey.getEncryptionKey());
-    const getEncryptionKey = opgpKey.getEncryptionKey.bind(opgpKey) as (keyid?: OpenPGP.Keyid | null, date?: Date, userId?: OpenPGP.UserId | null) => Promise<OpenPGP.key.Key | OpenPGP.key.SubKey | null>;
+    const getEncryptionKey = opgpKey.getEncryptionKey.bind(opgpKey) as
```

Comment:
Is there a way to achieve the same without using `.bind`? I'm trying to stay away from behavior that is too dynamic. Maybe it's just because I don't understand bind enough. But there is a preference for "dumb" code whenever possible, so that even I can read it :)

Reply:
@tomholub Yes, actually in some other programming languages (C#) when you pass a reference to a method (opgpKey.getEncryptionKey) this reference actually includes `this` reference for the non-static method (`opgpKey`). For Javascript, I have to bind `opgpKey` explicitly so that `getEncryptionKey` will see it as `this`.
Of course we can bind it via lambda like `getEncryptionKey = (params) => opgpKey.getEncryptionKey(params)` if you're more comfortable with this programming-style binding.

Reply:
Yes that definitely looks better. I think it's really just a matter of being comfortable with a certain style, this way it will be the same as the rest of codebase.

Reply:
I thought I have forbidden `.bind` with a linter, but it looks like I didn't. I'll make another issue.

Reply:
> 
> You could still -encrypt- for such -public- key if you pretend to encrypt at an older date. We have such mechanism where user can choose to do it, after they confirm a warning that it's not recommended.

Ok then, I'll restore `usableButExpired` calculation. I just derived that logic from key-import-ui.

> 
> Sure, maybe you want to verify old signatures made by corresponding private key.

`usableButExpired` is deduced based on encryption key only (signing key isn't taken into consideration). Maybe it's time to introduce 2 properties: `usableForEncryptionButExpired` and `usableForSigningButExpired`?